iothub/device/src/WritablePropertyResponse.cs

```diff
@@ -6,30 +6,72 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// The default implementation of the <see cref="WritablePropertyBase"/> that uses <see cref="JsonPropertyAttribute"/> to define the JSON property names.
+    /// An optional, helper class for constructing a writable property response.
     /// </summary>
-    public sealed class WritablePropertyResponse : WritablePropertyBase
+    /// <remarks>
+    /// This helper class will only work with <see cref="Newtonsoft.Json"/>.
+    /// It uses <see cref="Newtonsoft.Json"/> based <see cref="JsonPropertyAttribute"/> to define the JSON property names.
+    /// For scenarios where you want to use a different serializer, you will need to implement one with its corresponding seriailizer attributes.
+    /// </remarks>
+    public sealed class WritablePropertyResponse
     {
-        /// <inheritdoc />
-        public WritablePropertyResponse(object propertyValue, int ackCode, long ackVersion, string ackDescription = default)            
-            : base(propertyValue, ackCode, ackVersion, ackDescription)
+        /// <summary>
+        /// Represents the JSON document property name for the value
+        /// </summary>
+        public const string ValuePropertyName = "value";
+
+        /// <summary>
+        /// Represents the JSON document property name for the Ack Code
+        /// </summary>
+        public const string AckCodePropertyName = "ac";
+
+        /// <summary>
+        /// Represents the JSON document property name for the Ack Version
+        /// </summary>
+        public const string AckVersionPropertyName = "av";
+
+        /// <summary>
+        /// Represents the JSON document property name for the Ack Description
+        /// </summary>
+        public const string AckDescriptionPropertyName = "ad";
+
+        /// <summary>
+        /// Convenience constructor for specifying the properties.
+        /// </summary>
```

Comment:
Add an <example> tag with a <code> block to show what the System.Text.Json version would look like. This will make sure that if someone comes looking at this they will understand the class is used in the convention and that it will need to be properly serailized.